custom_components/elevenlabs_tts/__init__.py

```diff
@@ -1,33 +1,53 @@
 """ElevenLabs TTS Custom Integration"""
 import logging
 
-from homeassistant import core
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import ConfigEntryNotReady
 import requests
 
-from .const import CONF_VOICE, DEFAULT_VOICE
+from .const import DEFAULT_VOICE, DOMAIN, PLATFORMS
 from .elevenlabs import ElevenLabsClient
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_setup_entry(hass: core.HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up the ElevenLabs TTS component from a config entry."""
-    voice_name = entry.data.get(CONF_VOICE, DEFAULT_VOICE)
-    client = ElevenLabsClient(entry.data)
+    hass.data.setdefault(DOMAIN, {})
+
+    client = ElevenLabsClient(hass, entry)
+
+    hass.data[DOMAIN][entry.entry_id] = client
 
     try:
-        await hass.async_add_executor_job(client.get_voices)
+        await client.get_voices()
```

Comment:
because of the changes to the client (that it catches and logs errors, and returns empty responses) this should be 
```python
    voices = await client.get_voices()
    if not voices:
        return False
```